### PR TITLE
Refs docs

### PR DESCRIPTION
**Pull request by DavieReid**

### CODE REVIEW

File: docs/author/frontmatter.mdx [hunk 1]
```diff
@@ -72,6 +72,70 @@ async function $afterSource(pages, { config, ignorePages, pageExtensions }) {
...
 
-Todo
+Let's take a look at the `SharedConfigPlugin`.
+
+The purpose of this plugin is to crawl the page hierarchy to find the closest `sharedConfig` found in any parent page's frontmatter.
+
+- Finds all index pages among the source docs
+- Deserialises those pages so it can read the frontmatter and content of the page
+- If a property called `sharedConfig` in the page frontmatter a new files named shared-config.json is created
+- Sets a ref
```
Comment (DavieReid) [inline docs/author/frontmatter.mdx:84]: Need to complete this...

File: docs/author/refs.mdx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+The value of `data.sidebarPriority` comes from `sidebar.priority` in the frontmatter.
+
+<Callout>
+  Notice that because we don't specify a path before the `#` in the ref we need to put the whole
+  value inside quotes.
+</Callout>
+
+## Remote Refs (Cross-schema reference)
+
+It is possible to reference frontmatter of other pages. Again this helps avoid duplication but the real power is using refs to build the data model. See [advanced](./advanced) for more information.
```
[the remark this answers is not in the document]
Comment (DavieReid) [reply docs/author/refs.mdx:54]: DONE

File: docs/author/frontmatter.mdx (hunk 1 above)
Comment (DavieReid) [reply docs/author/frontmatter.mdx]: DONE

**Comment by lilyvc** [comment]
> Need a way to set the sidebar for the e2e test pages. As docs are being added the sidebar is being changed causing the snapshot test to fail. Any ideas about the best way to go about this? CC @lilyvc @mark-tate

once we update the sidebar rootDirGlob for the site, the test pages will not have other site pages in the sidebar so it won't keep changing. raising this PR now

File: packages/site/e2e/layout.detailHighlight.test.ts-snapshots/DetailHighlight-Layout-Desktop-Chrome-darwin.png
Comment (tomhazledine) [inline packages/site/e2e/layout.detailHighlight.test.ts-snapshots/DetailHighlight-Layout-Desktop-Chrome-darwin.png:1]: Will we get diffs like this in all PRs from now on? Does playwright regen the snapshots every time it runs?

**Review by lilyvc — APPROVED** [review]

File: packages/site/e2e/layout.detailHighlight.test.ts-snapshots/DetailHighlight-Layout-Desktop-Chrome-darwin.png
Comment (DavieReid) [reply packages/site/e2e/layout.detailHighlight.test.ts-snapshots/DetailHighlight-Layout-Desktop-Chrome-darwin.png:1]: no it doesn't regen the snapshot unless you run ` npx playwright test --update-snapshots`.

I am looking to relax diff level or run the tests in docker...

**Pull request merged**